src/main/scala/funsets/FunSets.scala

```diff
@@ -55,23 +55,34 @@ trait FunSets extends FunSetsInterface {
    */
   def forall(s: FunSet, p: Int => Boolean): Boolean = {
     def iter(a: Int): Boolean = {
-      if (???) ???
-      else if (???) ???
-      else iter(???)
+      if (a > bound) true
+      else if (contains(s,a) && !p(a)) false
+      else iter(a + 1)
     }
-    iter(???)
+    iter(-bound)
   }
 
   /**
    * Returns whether there exists a bounded integer within `s`
    * that satisfies `p`.
    */
-  def exists(s: FunSet, p: Int => Boolean): Boolean = ???
+  def exists(s: FunSet, p: Int => Boolean): Boolean = {
+   !forall(s, x => !p(x))
+  }
 
   /**
    * Returns a set transformed by applying `f` to each element of `s`.
    */
-  def map(s: FunSet, f: Int => Int): FunSet = ???
+  def map(s: FunSet, f: Int => Int): FunSet =
+    {
+      def iter(a: Int): FunSet = {
+        if (a > bound) null
+        else if (exists(s, x => x == a) ) singletonSet(f(a))
+        else iter(a + 1)
+      }
+      iter(-bound)
+
+    }
```

Comment:
Are these two alternative solutions correct?
- `x => exists(s, y => x == f(y))`
- `x => s(f(x))`

Reply:
sono entrambe corrette però nella seconda non viene utilizzato l'exist come richesto dall'esercizio , la prima invece è equivalente a questa giusto? : (x : Int) => exists(s, y => x == f(y)) 
devo prendere ancora confidenza con vari modi di scrivere in scala